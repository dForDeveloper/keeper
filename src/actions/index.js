export const setNotes = (notes) => ({
  type: 'SET_NOTES',
  notes
});

export const addNote = (note) => ({
  type: 'ADD_NOTE',
  note
});

export const updateNote = (note) => ({
  type: 'UPDATE_NOTE',
  note
});

export const deleteNote = (id) => ({
  type: 'DELETE_NOTE',
  id
});

export const toggleLoading = (bool) => ({
  type: 'TOGGLE_LOADING',
  bool
});

export const setError = (message) => ({
  type: 'SET_ERROR',
  message
});

export const setStatus = (code) => ({
  type: 'SET_STATUS',
  code
});

export const setQuery = (query) => ({
  type: 'SET_QUERY',
  query
});

export const setUser = (user) => ({
  type: 'SET_USER',
  user
});