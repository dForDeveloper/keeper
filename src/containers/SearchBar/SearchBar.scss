.SearchBar--input {
  background-color: #f5f5f5;
  padding: 8px 20px;
  border-radius: 5px;
  display: inline-block;
  width: 230px;
  border: none;
  outline: none;
  text-align: left;
  
  &::placeholder {
    color: #707070;
    opacity: .4;
  }
}