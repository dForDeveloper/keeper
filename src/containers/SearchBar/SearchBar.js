import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setQuery } from '../../actions';
import PropTypes from 'prop-types';

export class SearchBar extends Component {
  filterNotes = (e) => {
    const userQuery = e.target.value;
    const { setQuery } = this.props;
    setQuery(userQuery);
  }

  render() {
    return (
      <input
        onChange={this.filterNotes}
        className='SearchBar--input'
        placeholder='Search notes' />
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setQuery: (query) => dispatch(setQuery(query))
});

export default connect(null, mapDispatchToProps)(SearchBar);

SearchBar.propTypes = {
  setQuery: PropTypes.func
}