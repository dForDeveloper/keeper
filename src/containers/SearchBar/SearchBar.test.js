import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar, mapDispatchToProps } from '../SearchBar/SearchBar';
import { setQuery } from '../../actions';

describe('SearchBar', () => {
  describe('SearchBar component', () => {
    let wrapper;
    const mockSetQuery = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<SearchBar setQuery={mockSetQuery} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call setQuery with the query on change', () => {
      const mockEvent = { target: { value: 'hey' } };
      wrapper.find('.SearchBar--input').simulate('change', mockEvent);
      expect(mockSetQuery).toHaveBeenCalledWith('hey');
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with setQuery', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setQuery('note to find');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setQuery('note to find');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
}); 