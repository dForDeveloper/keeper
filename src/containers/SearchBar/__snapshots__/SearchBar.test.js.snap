// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar component should match the snapshot 1`] = `
<input
  className="SearchBar--input"
  onChange={[Function]}
  placeholder="Search notes"
/>
`;
